R/Rules-methods.R

```diff
@@ -4,127 +4,133 @@
+#'   named `singlePlots` is included. The `singlePlots` is itself a list with
+#'   single plots. An additional list with elements describing the outcome
+#'   of the rule can be contained too.
+#'
+#' @export
+#'
+setGeneric(
+  name = "nextBest",
+  def = function(nextBest,
+                 doselimit,
+                 samples,
+                 model,
+                 data,
+                 ...) {
+    if (!missing(doselimit) && length(doselimit) != 0) {
+      assert_number(doselimit, lower = 0, finite = TRUE)
+    }
+    standardGeneric("nextBest")
+  },
+  valueClass = "list"
+)
 
-              ## but now, round to the next possible grid point
-              index <- which.min(abs(data@doseGrid[dosesOK] - mtdEstimate))
-              ret <- data@doseGrid[dosesOK][index]
+## NextBestMTD ----
 
-              ## produce plot
-              plot1 <- ggplot() +
-                      geom_density(data=
-                                   data.frame(x=mtd_samples),
-                                   aes(x=x),
-                                   fill = "grey50", colour = "grey50") +
-                          xlab("MTD") + ylab("Posterior density") +
-                              xlim(range(data@doseGrid))
+#' @describeIn nextBest find the next best dose based on the MTD rule.
+#'
+#' @aliases nextBest-NextBestMTD
+#'
+#' @export
+#' @example examples/Rules-method-NextBestMTD.R
+#'
+setMethod(
+  f = "nextBest",
+  signature = signature(
+    nextBest = "NextBestMTD",
+    doselimit = "numeric",
+    samples = "Samples",
+    model = "Model",
+    data = "Data"
+  ),
+  definition = function(nextBest,
+                        doselimit,
+                        samples,
+                        model,
+                        data,
+                        ...) {
+
+    # Which doses on grid are <= maximum possible dose (if specified).
+    doseGrid_leq_limit <- if (length(doselimit) == 0) {
+      warning("doselimit is not provided, therefore no dose limit will be applied")
+      data@doseGrid
+    } else {
+      data@doseGrid[data@doseGrid <= doselimit]
+    }
 
-              plot1 <- plot1 +
-                  geom_vline(xintercept=mtdEstimate, colour="black", lwd=1.1) +
-                      geom_text(data=
-                                data.frame(x=mtdEstimate),
-                                aes(x, 0,
-                                    label = "Est", hjust=+1, vjust = +1),
-                                colour="black")
+    # Generate the MTD samples and derive the next best dose.
+    mtd_samples <- dose(x = nextBest@target, model, samples)
+    mtd_estimate <- nextBest@derive(mtd_samples = mtd_samples)
+    # Round to the next possible grid point.
+    min_dist_ind <- which.min(abs(doseGrid_leq_limit - mtd_estimate))
+    ret <- doseGrid_leq_limit[min_dist_ind]
+
+    # Create a plot.
+    p <- ggplot(
+      data = data.frame(x = mtd_samples),
+      aes(x = x),
```

Comment:
I guess we need
```suggestion
      aes(x = .data$x),
```
to avoid `R CMD check` note, also for the other aesthetics. See e.g. https://github.com/insightsengineering/hermes/blob/ce68571f3ccac0fa17945587fb587f911f604c81/R/draw_barplot.R#L86 as example